sdk/servicebus/azure-servicebus/tests/servicebus_preparer.py

```diff
@@ -212,15 +213,25 @@ def create_resource(self, name, **kwargs):
             group = self._get_resource_group(**kwargs)
             namespace = self._get_namespace(**kwargs)
             topic = self._get_topic(**kwargs)
-            self.resource = self.client.subscriptions.create_or_update(
-                group.name,
-                namespace.name,
-                topic.name,
-                name,
-                SBSubscription(
-                    requires_session=self.requires_session
-                )
-            )
+
+            retries = 4
+            for i in range(retries):
+                try:
+                    self.resource = self.client.subscriptions.create_or_update(
+                        group.name,
+                        namespace.name,
+                        topic.name,
+                        name,
+                        SBSubscription(
+                            requires_session=self.requires_session
+                        )
+                    )
+                    break
+                except Exception as ex:
+                    error = "The requested resource {} does not exist".format(namespace)
```

Comment:
I'm not sure about what `str(ex)` would give us -- based on the pipeline running log, it seems to be two level.
1)
```
Exception: Preparer failure when creating resource ServiceBusQueueAuthorizationRulePreparer for test test_sb_client_incorrect_queue_conn_str (test_sb_client.ServiceBusClientTests): Operation returned an invalid status code 'Not Found'
```
2)
```
Detailed error message: The requested resource servicebustest_qtwoxdnfo does not exist.
```

would `str(ex)` give us both or just one of them?
one potential way I'm thinking is to be a bit defensive, we also check the str `Operation returned an invalid status code 'Not Found'`.